tools/build-toolchain.sh

```diff
@@ -1,34 +1,71 @@
 #! /bin/bash
-# N64 MIPS GCC toolchain build/install script for Unix distributions
+# N64 MIPS GCC toolchain build/install script.
 # (c) 2012-2021 Shaun Taylor and libDragon Contributors.
 # See the root folder for license information.
 
 
+# This script may prompt for a password if it is not called with elevated privileges.
+# Preferably it is fixed at the start!
+#TODO: ensure elevated privileges! e.g.
+#echo "$(whoami)"
+#[ "$UID" -eq 0 ] || exec sudo "$0" "$@"
 
-# Before calling this script, make sure you have GMP, MPFR and TexInfo
+# Before calling this script, make sure you have all required dependency
 # packages installed in your system.  On a Debian-based system this is
 # achieved by typing the following commands:
 #
-# sudo apt-get install libmpfr-dev
-# sudo apt-get install texinfo
-# sudo apt-get install libmpc-dev
+# sudo apt-get update && sudo apt-get upgrade
+# sudo apt-get install -yq wget bzip2 gcc g++ make file libmpfr-dev libmpc-dev zlib1g-dev texinfo git gcc-multilib
 
 # Exit on error
 set -e
 
-# Set N64_INST before calling the script to change the default installation directory path
-INSTALL_PATH="${N64_INST:-/usr/local}"
+# Check for cross compile script flag
+if [ "$1" == "-xcw" ]; then # Windows cross compile flag is specified as a parameter.
+  # This (probably) requires the toolchain to have already built and installed on the native (linux) system first!
+  # This (may) also require the following (extra) package dependencies:
+  # sudo apt-get install -yq mingw-w64 libgmp-dev bison libz-mingw-w64-dev autoconf
+
+  echo "Cross compiling for Windows x86_64"
+  # Use the current-directory/win_binaries for the install path, as this is not for linux!
+  rm -rf win_binaries
+  mkdir win_binaries
+  CURRENT_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+  INSTALL_PATH="$CURRENT_PATH/win_binaries"
```

Comment:
This is because we want to use the `build-toolchain.sh`'s current directory as a basis for the install location for packaging and it should definitely not be installing any of the output to the current N64INST location... However it might be better to handle it in the github workflow script (although it will be harder to handle that for users who want to handle it). Although, it should be possible to combine them now, as they were originally split to handle MinGW libs.